internal/fs/dir_handle_test.go

```diff
@@ -0,0 +1,272 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fs
+
+import (
+	"os"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/fs/inode"
+	"github.com/googlecloudplatform/gcsfuse/internal/gcsx"
+	"github.com/jacobsa/fuse/fuseops"
+	"github.com/jacobsa/gcloud/gcs"
+	"github.com/jacobsa/gcloud/gcs/gcsfake"
+	. "github.com/jacobsa/ogletest"
+	"github.com/jacobsa/timeutil"
+	"golang.org/x/net/context"
+)
+
+const dirInodeID = 17
+const dirInodeName = "foo/"
+const dirMode os.FileMode = 0712 | os.ModeDir
+const fileUnderDir = "bar"
+const implicitDirName = "baz"
+const typeCacheTTL = time.Second
+const uid = 123
+const gid = 456
+const tmpObjectPrefix = ".gcsfuse_tmp/"
+const appendThreshold = 1
+const fakeBucketName = "some_bucket"
+const dummyInodeId = fuseops.InodeID(2)
+const dummyfuseid = uint64(4)
+const dummypid = uint32(6)
+const dummyHandleId = fuseops.HandleID(8)
+
+type DirHandleTest struct {
+	ctx    context.Context
+	bucket gcsx.SyncerBucket
+	clock  timeutil.SimulatedClock
+	dh     *dirHandle
+}
+
+var _ SetUpInterface = &DirHandleTest{}
+var _ TearDownInterface = &DirHandleTest{}
+
+func init() {
+	RegisterTestSuite(&DirHandleTest{})
+}
+
+func (t *DirHandleTest) SetUp(ti *TestInfo) {
+	t.ctx = ti.Ctx
+	t.clock.SetTime(time.Date(2023, 6, 26, 15, 55, 0, 0, time.Local))
```

Comment:
can't you set current time. e.g. time.now?

Reply:
Done.